### PR TITLE
feat: add files option

### PR DESCRIPTION
**Pull request by chihab**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #6

## What is the new behavior?

New files option where we can put  the list of file patterns containing `_BUILD_HASH_`

Example:
```sh
ng run <your-app>:stamp --files=*.js,*.json
```

Also configurable from builder options in angular|workspace.json

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by manekinekko — APPROVED** [review]
LGTM

File: packages/version-stamp/src/lib/stamp/schema.json [hunk 1]
```diff
@@ -27,7 +27,12 @@
...
     "outputPath": {
       "type": "string",
-      "description": "The version to stamp",
+      "description": "The output path",
+      "$valueDescription": "configuration"
+    },
+    "files": {
+      "type": "array",
+      "description": "The list of patterns to match the files with",
       "$valueDescription": "configuration"
```
Comment (Jefiozie) [inline packages/version-stamp/src/lib/stamp/schema.json]:
```suggestion
    "files": {
      "type": "array",
      "description": "The list of patterns to match the files with",
      "$valueDescription": "configuration",
      "default":  ["*.js", "*.json"]
```

File: packages/version-stamp/src/lib/stamp/index.ts [hunk 2]
```diff
@@ -38,6 +38,10 @@ export default createBuilder<any>(
       let buildNumber = options.version;
 
       if (result.success) {
+        options = {
+          ...options,
+          files: options.files.length ? options.files : ['*.js', '*.json'],
```
Comment (Jefiozie) [inline packages/version-stamp/src/lib/stamp/index.ts]:
```suggestion
          files: options.files,
```

File: workspace.json [hunk 3]
```diff
@@ -66,7 +66,8 @@
         "stamp": {
           "builder": "@xlayers/version-stamp:stamp",
           "options": {
-            "outputPath": "dist/packages/demo"
+            "outputPath": "dist/packages/demo",
+            "files": ["*.js", "*.json"]
```
Comment (Jefiozie) [inline workspace.json]:
```suggestion
```

**Review by Jefiozie — CHANGES_REQUESTED** [review]
Hey @chihab, thank you so much for this PR! I've made some small suggestions could you have a look at it?
I would like to lean on the defaults we can use from the schema, would you agree on this?

Let me know what you think about it and again thanks for using this lib and providing us with this PR and feedback. 🙏

**Review by Jefiozie — APPROVED** [review]
LGTM

**Pull request merged**